demos/blocklyfactory/block_exporter_tools.js

```diff
@@ -221,3 +210,61 @@ BlockExporterTools.prototype.generateToolboxFromLibrary
   xmlDom.appendChild(categoryElement);
   return xmlDom;
 };
+
+/**
+ * Generate xml for the workspace factory's category from imported block
+ * definitions.
+ *
+ * @return {!Element} Xml representation of a category.
+ */
+BlockExporterTools.prototype.generateCategoryFromBlockLib = function() {
+  // Create category DOM element.
+  var categoryElement = goog.dom.createDom('category');
+  categoryElement.setAttribute('name', 'Block Library');
+
+  var allBlockTypes = blockLibStorage.getBlockTypes();
+  // Object mapping block type to XML.
+  var blockXmlMap = blockLibStorage.getBlockXmlMap(allBlockTypes);
+
+  // Define the custom blocks in order to be able to create instances of
+  // them in the exporter workspace.
+  this.addBlockDefinitions(blockXmlMap);
+
+  for (var blockType in blockXmlMap) {
+    // Get block.
+    var block = FactoryUtils.getDefinedBlock(blockType, this.hiddenWorkspace);
+
+    // Get preview block XML.
+    var blockChild = Blockly.Xml.blockToDom(block);
+    blockChild.removeAttribute('id');
+
+    // Add block to category and category to XML.
+    categoryElement.appendChild(blockChild);
+  }
+  return categoryElement;
+};
+
+/**
+ * Given two lists of blocktypes, will return the list of blockTypes that
+ * are found in both lists.
+ * @private
+ *
+ * @param {!Array.<string>} blockTypesOne - types of blocks.
+ *    workspace from Workspace FactoryTwo.
+ * @param {!Array.<string>} blockTypes - array of types of blocks.
+ * TODO(quachtina96): Use as helper function in handler for  'Add All Used
+ *    Blocks' Button, to figure out which blocks used in the workspace factory
+ *    are stored in the block library. Probably more efficient if blockTypesTwo
+ *    is the block types stored in library.
+ */
+BlockExporterTools.prototype.getSharedTypes_ = function(blockTypesOne, blockTypesTwo) {
```

Comment:
This isn't used yet, right?  Since this PR is already big, can you leave it out for now and either add in the one where you're going to use it, or in a separate one?